internal/controller/templatechain_controller.go

```diff
@@ -154,7 +154,10 @@ func (r *TemplateChainReconciler) ReconcileTemplateChain(ctx context.Context, te
 				return ctrl.Result{}, fmt.Errorf("type assertion failed: expected ServiceTemplate but got %T", source)
 			}
 			spec := serviceTemplate.Spec
-			spec.Helm = kcm.HelmSpec{ChartRef: serviceTemplate.Status.ChartRef}
+			if spec.Helm == nil {
+				continue
+			}
```

Comment:
If we continue if `spec.Helm == nil` then the template will not be created below, even if .spec.Kustomize or .spec.Resources have been specified.

Reply:
Also do we need to do similar for ClusterTemplate case?